server/plugin/auth_token_test.go

```diff
@@ -0,0 +1,169 @@
+			description:   "Failed to create oauth token because io.ReadFull gives error",
+			expectedError: "failed to create auth token: mockError",
+			resdFullError: errors.New("mockError"),
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p := Plugin{}
+
+			p.setConfiguration(
+				&configuration{
+					EncryptionSecret: "mockEncryptionSecret",
+				})
+
+			monkey.Patch(aes.NewCipher, func(a []byte) (cipher.Block, error) {
+				return &mockBLock{}, testCase.newCipherError
+			})
+
+			monkey.Patch(cipher.NewGCM, func(_ cipher.Block) (cipher.AEAD, error) {
+				return &mockAesgcm{}, testCase.newGCMError
+			})
+
+			monkey.Patch(io.ReadFull, func(_ io.Reader, _ []byte) (int, error) {
+				return 1, testCase.resdFullError
+			})
+
+			tok := &oauth2.Token{}
+
+			res, err := p.NewEncodedAuthToken(tok)
+
+			if testCase.expectedError != "" {
+				require.EqualError(t, err, testCase.expectedError)
+				require.EqualValues(t, "", res)
+			} else {
+				require.Nil(t, err)
+				require.NotEqualValues(t, "", res)
+			}
+		})
+	}
+}
+
+func Test_ParseAuthToken(t *testing.T) {
+	defer monkey.UnpatchAll()
+
+	for _, testCase := range []struct {
+		description    string
+		expectedError  string
+		newCipherError error
+		newGCMError    error
+		encodedToken   string
+		unmarshalError error
+	}{
+		{
+			description:  "OAuth2 token is parsed successfully",
+			encodedToken: "mockEncodedToken",
+		},
+		{
+			description:    "Failed to decode oauth token because aes.NewCipher gives error",
+			expectedError:  "mockError",
+			newCipherError: errors.New("mockError"),
+			encodedToken:   "mockEncodedToken",
+		},
+		{
+			description:   "Failed to decode oauth token because cipher.NewGCM gives error",
+			expectedError: "mockError",
+			newGCMError:   errors.New("mockError"),
+			encodedToken:  "mockEncodedToken",
+		},
+		{
+			description:   "Failed to decode oauth token because token is too short",
+			expectedError: "token too short",
+			encodedToken:  "",
+		},
+		{
+			description:    "Failed to decode oauth token because json.Unmarshal gives error",
+			expectedError:  "mockError",
+			unmarshalError: errors.New("mockError"),
+			encodedToken:   "mockEncodedToken",
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p := Plugin{}
+
+			p.setConfiguration(
+				&configuration{
+					EncryptionSecret: "mockEncryptionSecret",
+				})
+
+			monkey.Patch(aes.NewCipher, func(a []byte) (cipher.Block, error) {
+				return &mockBLock{}, testCase.newCipherError
+			})
+
+			monkey.Patch(cipher.NewGCM, func(_ cipher.Block) (cipher.AEAD, error) {
+				return &mockAesgcm{}, testCase.newGCMError
+			})
+
+			monkey.Patch(json.Unmarshal, func(_ []byte, _ interface{}) error {
+				return testCase.unmarshalError
+			})
+
+			res, err := p.ParseAuthToken(testCase.encodedToken)
+
+			if testCase.expectedError != "" {
+				require.EqualError(t, err, testCase.expectedError)
+				require.Nil(t, res)
+			} else {
+				require.Nil(t, err)
```

Comment:
here can u check for its response

Reply:
I don't think its possible for this case